src/Compilers/Core/Desktop/AnalyzerFileReference.InMemoryAssemblyLoader.cs

```diff
@@ -143,23 +146,82 @@ private static Assembly LoadCore(string fullPath)
             }
 
             /// <summary>
-            /// Handles the <see cref="AppDomain.AssemblyResolve"/> event when the requesting
-            /// assembly is one that we've loaded.
-            /// 
-            /// We assume that an assembly's dependencies can be found next to it in the file
-            /// system.
+            /// Handles the <see cref="AppDomain.AssemblyResolve"/> event.
             /// </summary>
             private static Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEventArgs args)
             {
                 if (args.RequestingAssembly == null)
                 {
-                    // We don't know who is requesting the load; don't try to satisfy the request.
+                    return ResolveForUnknownRequestor(args.Name);
```

Comment:
The CLR doesn't load the assembly. Instead, we load it when we call `Assembly.Load(byte[])` (see the `LoadCore` method).